magento/magento2
# Update price index if product is added to website

### Description (*)
In the scenario of multiple websites, if a product is created and added to one website. It works properly, shows on frontend etc. In case the same product is added to another website via an API call, then the product won't be visible due to missing price index. It will either require to do full price index + cache clean to make the product visible or some additional change e.g. price, in order to be processed by price indexer. This PR adds an mview subscription to catalog_product_website table to inject the product id to catalog_product_price_cl and enable it to be processed by incremental indexer.

From admin interface adding product to another website works, as there is a choice of from which website to copy the data from (or use default values), but in case of API call, the product <> website relation is just inserted (it does actually update fulltext and catalog_category at the moment).

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#31069

### Manual testing scenarios (*)
1. Set indexers in scheduled mode
2. Create at least 2 websites with store views
3. Create a product in one of the websites, with sufficient data that it will be shown on frontend.
4. Fire an API call to **rest/V1/products/{sku}/websites**, adding the product created in step 1 to another website
5. Wait for next incremental indexer to process changes
6. Check that the product is not visible in the website it was assigned to, due to missing price in catalog_product_index_price for the corresponding website

### Questions or comments


### Contribution checklist (*)
 - [x ] Pull request has a meaningful description of its purpose
 - [x ] All commits are accompanied by meaningful commit messages


### Resolved issues:
1. [x] resolves magento/magento2#31069: Update price index if product is added to website
